### ISSUE TITLE
write_data from utils library have a negative side effect

### ISSUE THREAD

**Issue by JoseRuedaRueda** [D3DEnergetic/FIDASIM#287]
# General description
write_data from the utils library modifies the input data transposing it. Which is not nice if the user wants to keep using that input data in python for other things

# Problem cause
https://github.com/D3DEnergetic/FIDASIM/blob/1010753114b28589bc5666dc3d2ccaa121a3c862/lib/python/fidasim/utils.py#L763

# Changes
A routine that write data should never modify user input. So create a copy of dict and do everything with that dict copy

**Comment by lstagner** [comment]
The reason it does a transpose is because Python is row-major but Fortran/Julia are column major. If I don't do the transpose then Fortran cant read the files. That being said it may be a good idea to make that behavior a keyword argument.

**Comment by JoseRuedaRueda** [comment]
Yep, I understand the reason why we have to transpose before writing, my problem is the modification os the input with the user unaware

I will add a flag to all write routines: if true (default), transpose inputs as until now, if false, transpose stuff before writing the file, but leave input untouched
